cmd/remote-storage/app/server.go

```diff
@@ -143,3 +139,21 @@ func (s *Server) Close() error {
 	s.telset.ReportStatus(componentstatus.NewEvent(componentstatus.StatusStopped))
 	return nil
 }
+
+func initArchiveStorage(
```

Comment:
this looks redundant with query_service. Why not have that as a method of the `storage.Factory` struct itself?